ui/src/app/edge/live/common/selfconsumption/modal/modal.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { LINE_INFO } from "src/app/shared/edge/edgeconfig.spec";
+import { OeFormlyViewTester } from "src/app/shared/genericComponents/shared/tester";
+import { sharedSetup, TestContext } from "src/app/shared/test/utils.spec";
+
+import { ModalComponent } from "./modal";
+
+export const VIEW_CONTEXT: OeFormlyViewTester.Context = ({});
+
+export function expectView(testContext: TestContext, viewContext: OeFormlyViewTester.Context, view: OeFormlyViewTester.View): void {
+  expect(OeFormlyViewTester.apply(ModalComponent.generateView(testContext.translate), viewContext))
+    .toEqual(view);
+};
+
+describe('SelfConsumption - Modal', () => {
+  let TEST_CONTEXT;
+  beforeEach(() => TEST_CONTEXT = sharedSetup());
+
+  it('generateView()', () => {
+    {
+      expectView(TEST_CONTEXT, VIEW_CONTEXT, {
+        title: "Eigenverbrauch",
+        lines: [
+          LINE_INFO("Der Eigenverbrauch gibt an zu wie viel Prozent die aktuell erzeugte Leistung durch direkten Verbrauch und durch Speicherbeladung selbst genutzt wird.")
+        ]
+      });
```

Comment:
here as well. similar to the one above. it's a good practice to separate the object and store it in a variable. so that it's easier to read what is expected.

Reply:
I want to do that for next PR